compose/foundation/foundation/src/desktopMain/kotlin/androidx/compose/foundation/text/KeyMapping.desktop.kt

```diff
@@ -110,8 +110,8 @@ internal actual val platformDefaultKeyMapping: KeyMapping =
 
                         event.isShiftPressed ->
                             when (event.key) {
-                                MappedKeys.MoveHome -> KeyCommand.SELECT_HOME
-                                MappedKeys.MoveEnd -> KeyCommand.SELECT_END
+                                MappedKeys.MoveHome -> KeyCommand.SELECT_LINE_START
```

Comment:
Should we change it in `defaultKeyMapping` as well? I don't have a keyboard with Home button right now, so I can't verify it on Windows and Android.

Reply:
Actually, I can press Fn + Left Arrow ).

On macOs we select the whole text, not the whole line. Tested in TextEdit and in Pages

Reply:
I tested in google docs (mac os), and there they select the line only. And in IDEA on mac os, it selects only the line.

What do you think, Should we make this behaviour dependant on a platform?

Reply:
Checked on Windows - when we press Home/End, we move to the start/end of the line. If we press Shift + Home/End - we select part of the line

Reply:
> What do you think, Should we make this behaviour dependant on a platform?

I think, key behaviour should mimic the native behaviour.

Google Docs and IDEA probably not best role models :).

Best role models are native apps, then, probably Chromium. In TextArea on macOs we have the native behaviour.

Reply:
The current not-spoken strategy was to mimic feel on native platform, but not mimic look.

Reply:
Update the PR. 
Window and Linux will select only the line where the cursor is.
MacOS will select till the end/start of the text.

The tests use reflection to mock the top level field `platformDefaultKeyMapping`.